test/test_helpers.h

```diff
@@ -105,6 +273,88 @@ inline void single_triangle_cw(const Vec3r<T> &center, Vec3rList<T> &vertices, V
   indices.emplace_back(Vec3ui{0, 1, 2});
 }
 
+// Creation of triangle in xy plane / clockwise
+template<typename T>
+inline void single_triangle_cw_flat_xy(const Vec3r<T> &center, Vec3rList<T> &vertices, Vec3iList &indices) {
+  // v1 = (0,0,0), v2 = (1,0,0), v3 = (0,1,0)
+  vertices.emplace_back(Vec3r<T>{center[0] + T(0), center[1] + T(0), center[2] + T(0)});
+  vertices.emplace_back(Vec3r<T>{center[0] + T(1.), center[1] + T(0), center[2] + T(0)});
+  vertices.emplace_back(Vec3r<T>{center[0] + T(0), center[1] + T(1.), center[2] + T(0)});
+
+  indices.emplace_back(Vec3ui{0, 1, 2});
+}
+
+// Creation of triangle in xz plane / clockwise
+template<typename T>
+inline void single_triangle_cw_flat_xz(const Vec3r<T> &center, Vec3rList<T> &vertices, Vec3iList &indices) {
+  // 0  (0,0,0) + center
+  vertices.emplace_back(Vec3r<T>{center[0], center[1], center[2]});
+  // 1  (1, 0, 0) + center
+  vertices.emplace_back(Vec3r<T>{center[0] + T(1.), center[1], center[2]});
+  // 2  (0, 0, 1) + center
+  vertices.emplace_back(Vec3r<T>{center[0], center[1], center[2] + T(1.)});
+
+  indices.emplace_back(Vec3ui{0, 1, 2});
+}
+
+// Creation of triangle in yz plane / clockwise
+template<typename T>
+inline void single_triangle_cw_flat_yz(const Vec3r<T> &center, Vec3rList<T> &vertices, Vec3iList &indices) {
+  // 0  (0, 0, 0) + center
+  vertices.emplace_back(Vec3r<T>{center[0], center[1], center[2]});
+  // 1  (0, 1, 0) + center
+  vertices.emplace_back(Vec3r<T>{center[0], center[1] + T(1.), center[2]});
+  // 2  (0, 0, 1) + center
+  vertices.emplace_back(Vec3r<T>{center[0], center[1], center[2] + T(1.)});
+
+  indices.emplace_back(Vec3ui{0, 1, 2});
+}
+
+// Creation of triangle in xy plane / counter clockwise
+template<typename T>
+inline void single_triangle_ccw_flat_xy(const Vec3r<T> &center, Vec3rList<T> &vertices, Vec3iList &indices) {
+  // v1 = (0,0,0), v2 = (1,0,0), v3 = (0,1,0)
+  vertices.emplace_back(Vec3r<T>{center[0], center[1], center[2]});
+  vertices.emplace_back(Vec3r<T>{center[0] + T(1.), center[1], center[2]});
+  vertices.emplace_back(Vec3r<T>{center[0], center[1] + T(1.), center[2]});
+
+  indices.emplace_back(Vec3ui{0, 2, 1});
+}
+
+/**
+ * Pyramid mesh
```

Comment:
please provide a little more info about the pyramid (e.g. orientation in R^3)

Reply:
done